tests/Admin/AdminTest.php

```diff
@@ -1731,6 +1732,35 @@ public function testGetSubject($id): void
         static::assertSame($model, $admin->getSubject()); // model manager must be used only once
     }
 
+    public function testSetSubject(): void
+    {
+        $this->expectNotToPerformAssertions();
+
+        $admin = new PostAdmin();
+        $admin->setModelClass(Post::class);
+        $admin->setModelManager($this->createMock(ModelManagerInterface::class));
+
+        $admin->setSubject(new BlogPost());
+        $admin->setSubject(new NewsPost());
+        $admin->setSubject(new Post());
```

Comment:
These both would fail before:

```
LogicException: Admin "Sonata\AdminBundle\Tests\Fixtures\Admin\PostAdmin" does not allow this subject: Sonata\AdminBundle\Tests\Fixtures\Bundle\Entity\NewsPost, use the one register with this admin class Sonata\AdminBundle\Tests\Fixtures\Bundle\Entity\BlogPost
```